### PR TITLE
CBG-940 Only invoke Complete() for completion of one-shot replications

### PR DESCRIPTION
**Pull request by adamcfraser**
Avoids replications going to a stopped state when they are rebalanced to another node.

### CODE REVIEW

File: rest/replication_api_test.go [hunk 1]
```diff
@@ -405,3 +409,353 @@ func TestReplicationsFromConfig(t *testing.T) {
...
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyReplicate, base.KeyHTTP, base.KeyHTTPResp, base.KeySync, base.KeySyncMsg)()
+
+	activeRT, remoteRT, remoteURLString, teardown := setupSGRPeers(t)
+	defer teardown()
+
+	// Create docs on active
+	docABC1 := t.Name() + "ABC1"
+	docDEF1 := t.Name() + "DEF1"
+	_ = remoteRT.putDoc(docABC1, `{"source":"activeRT","channels":["ABC"]}`)
+	_ = remoteRT.putDoc(docDEF1, `{"source":"activeRT","channels":["DEF"]}`)
```
Comment (bbrks) [inline rest/replication_api_test.go]: These need to be created on `activeRT`, but when I change it, I'm seeing weird behaviour where `docABC2` isn't sent without a sleep on line 615

```suggestion
	_ = activeRT.putDoc(docABC1, `{"source":"activeRT","channels":["ABC"]}`)
	_ = activeRT.putDoc(docDEF1, `{"source":"activeRT","channels":["DEF"]}`)
```

**Review by bbrks — CHANGES_REQUESTED** [review]
All looks good, apart from the below comment.

File: rest/replication_api_test.go (hunk 1 above)
Comment (adamcfraser) [reply rest/replication_api_test.go]: Thanks for catching - I'll take a look.

Comment (adamcfraser) [reply rest/replication_api_test.go]: Made that fix, and added handling to avoid the race in CBG-1000 until it's fixed.

**Review by bbrks — APPROVED** [review]

**Pull request merged**